### PR TITLE
Clearing worker processor instances of reactor state after every interaction

### PR DESCRIPTION
**Pull request by zachmprince**
## What is the change?

The PR creates a method in `Operator` that is called `_finalizeInteract` which is called inside interface loop of `_interactAll`. This method is overridden by `OperatorMPI` to "reset" worker processors by broadcasting a command to force execution of `_resetWorker`.

## Why is the change being made?

The purpose of this change to force worker resets after each interaction is complete, instead of relying on plugins to return the correct `mpiActionRequiresReset` during `MpiAction` invocations. The goal was to reduce memory when going to subsequent interfaces that don't require a distributed state. However, the addition of this reset call is not reducing memory on worker processors, which means the "reset" implementation may be flawed. However, the change proposed in this PR is still appropriate in principle and improvements to the reset functionality can be improved in the future.

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [ ] The [release notes](https://terrapower.github.io/armi/developer/tooling.html#add-release-notes) have been updated if necessary.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

**Review by opotowsky — APPROVED** [review]
🚀

File: armi/operators/operatorMPI.py [hunk 1]
```diff
@@ -163,6 +163,8 @@ def workerOperate(self):
                 note = context.MPI_COMM.bcast("wait", root=0)
                 if note != "wait":
                     raise RuntimeError('did not get "wait". Got {0}'.format(note))
+            elif cmd == "reset":
```
Comment (john-science) [inline armi/operators/operatorMPI.py:166]: I actually think ARMI has _way_ too many log statements as it is.  People add them, forget they exist, and never look at them again. Happens all the time.

Why is this one necessary for people to see during ever run?

Comment (zachmprince) [reply armi/operators/operatorMPI.py:166]: Definitely helped with confirming that this implementation was working. This specific one is only outputted on worker nodes, but there is one in `_finalizeInteract` that is logged on root. I can get rid of both, one of them, or none of them. I'll leave that to your decision.

**Comment by john-science** [comment]
Quick Question:

You added new code, why are there no tests of this?  (Did you try, and it was too hard to test?)

File: armi/operators/operatorMPI.py (hunk 1 above)
Comment (zachmprince) [reply armi/operators/operatorMPI.py:166]: Indeed. I definitely want to avoid what's in the `else` block with this new command.

**Comment by zachmprince** [comment]
> Quick Question:
> 
> You added new code, why are there no tests of this? (Did you try, and it was too hard to test?)

I just didn't think of it. I can add a test.

File: armi/bookkeeping/memoryProfiler.py [hunk 2]
```diff
@@ -342,7 +342,7 @@ def __init__(self):
         self.processMemoryInMB: Optional[float] = None
         if _havePsutil:
             self.percentNodeRamUsed = psutil.virtual_memory().percent
-            self.processMemoryInMB = psutil.Process().memory_info().rss / (1012.0**2)
+            self.processMemoryInMB = psutil.Process().memory_info().rss / (1024.0**2)
```
Comment (john-science) [inline armi/bookkeeping/memoryProfiler.py:345]: lawls

**Comment by zachmprince** [comment]
@john-science @opotowsky Changed log message to `extra` and added a test in `test_mpiFeatures`.

**Review by john-science — APPROVED** [review]

**Pull request merged**